lib/basedriver/commands/session.js

```diff
@@ -15,34 +15,27 @@ commands.createSession = async function createSession (jsonwpDesiredCapabilities
 
   log.debug();
 
-  // Determine weather we should use jsonwpDesiredCapabilities or w3cCapabilities to get caps from
-  let caps;
-  if (w3cCapabilities) {
-    this.setProtocolW3C();
-    if (jsonwpDesiredCapabilities) {
-      log.debug(`W3C capabilities and MJSONWP desired capabilities were provided`);
-    }
+  if (!w3cCapabilities) {
+    throw new errors.SessionNotCreatedError('Appium only supports W3C capabilitites');
+  }
 
-    if (jsonwpDesiredCapabilities && !_.isPlainObject(w3cCapabilities)) {
-      // If W3C Capabilities and MJSONWP Capabilities were provided and W3C caps aren't a plain object,
-      // log a warning and fall back to MJSONWP
-      if (!_.isEmpty(w3cCapabilities)) {
-        log.warn(`Expected W3C "capabilities" to be a JSON Object but was provided with: ${JSON.stringify(w3cCapabilities)}`);
-      }
-      log.warn(`Falling back to MJSONWP desired capabilities`);
-      this.setProtocolMJSONWP();
-      caps = jsonwpDesiredCapabilities;
-    } else {
-      log.debug(`Creating session with W3C capabilities: ${JSON.stringify(w3cCapabilities, null, 2)}`);
-      caps = processCapabilities(w3cCapabilities, this.desiredCapConstraints, this.shouldValidateCaps);
-    }
-  } else {
-    this.setProtocolMJSONWP();
-    log.debug(`Creating session with MJSONWP desired capabilities: ${JSON.stringify(jsonwpDesiredCapabilities, null, 2)}`);
-    caps = jsonwpDesiredCapabilities || {};
+  if (jsonwpDesiredCapabilities) {
+    log.warn('Appium will ignore the given MJSONWP desired capabilities');
```

Comment:
`Appium received (M)JSONWP desired capabilities in alongside the W3C capabilities; they will be ignored`